mindsdb/api/mysql/mysql_proxy/classes/sql_query.py

```diff
@@ -1556,7 +1556,7 @@ def get_date_format(samples):
                     if isinstance(arg, Constant) and isinstance(arg.value, str):
                         arg.value = fnc(arg.value)
 
-        if self.model_types.get(order_col) in ('date', 'datetime'):
+        if self.model_types.get(order_col) in ('date', 'datetime') or isinstance(predictor_data[0][order_col], pd.Timestamp):  # noqa
```

Comment:
imho we should simplify this time filter urgently, but realistically that's another issue (part of the TS refactor).

As of now, this is a good compromise to avoid breaking anything.